src/string.cr

```diff
@@ -879,11 +905,53 @@ class String
     end
   end
 
-  def byte_slice(start : Int, count : Int)
+  # Returns a new string built from *count* bytes starting at *start* byte.
+  #
+  # *start* can can be negative to start counting
+  # from the end of the string.
+  # If *count* is bigger than the number of bytes from *start* to `#bytesize`,
+  # only remaining bytes are returned.
+  #
+  # This method should be avoided,
+  # unless the string is proven to be ASCII-only (for example `#ascii_only?`),
+  # or the byte positions are known to be at character boundaries.
+  # Otherwise the characters are splitted, which leads to invalid UTF-8 values.
+  #
+  # Raises `IndexError` if the *start* index is out of bounds.
+  #
+  # Raises `ArgumentError` if *count* is negative.
+  #
+  # ```
+  # "hello".byte_slice(0, 2)   # => "he"
+  # "hello".byte_slice(0, 100) # => "hello"
+  # "hello".byte_slice(-2, 3)  # => "he"
+  # "hello".byte_slice(-2, 5)  # => "he"
+  # "hello".byte_slice(-2, 5)  # => "he"
+  # "¥hello".byte_slice(0, 2)  # => "¥"
+  # "¥hello".byte_slice(2, 2)  # => "he"
+  # "¥hello".byte_slice(0, 1)  # => "�"
+  # "¥hello".byte_slice(1, 1)  # => "�"
+  # "¥hello".byte_slice(1, 2)  # => "�h"
+  # "hello".byte_slice(6, 2)   # raises IndexError
+  # "hello".byte_slice(-6, 2)  # raises IndexError
+  # "hello".byte_slice(0, -2)  # raises ArgumentError
+  # ```
+  def byte_slice(start : Int, count : Int) : String
     byte_slice?(start, count) || raise IndexError.new
   end
 
-  def byte_slice?(start : Int, count : Int)
+  # Like `byte_slice(Int, Int)` but returns `Nil` if the *start* index is out of bounds.
```

Comment:
I'd rather duplicate the entire documentation from `byte_slice` instead of referring to it. But I'm not sure what others think about it?

Reply:
(Eventually, it would be great if we could group similar methods together for the API docs and document them combined.)

Reply:
So? It's similar to other docs in std lib, for example Array# []? or Array#pop? or String#[]? and others. I think it would be sufficient to improve doc generator to turn these `` marks into html links.